annif/backend/nn_ensemble.py

```diff
@@ -129,9 +138,20 @@ def initialize(self, parallel: bool = False) -> None:
                 backend_id=self.backend_id,
             )
         self.debug("loading Keras model from {}".format(model_filename))
-        self._model = load_model(
-            model_filename, custom_objects={"MeanLayer": MeanLayer}
-        )
+        try:
+            self._model = load_model(
+                model_filename, custom_objects={"MeanLayer": MeanLayer}
+            )
+        except ValueError as err:
```

Comment:
Actually could this catch Exception, which arises from models trained on previous version of TF? I think some times there has been at least KeyError from load_model().

Reply:
Agree, catching Exception would guard against a wider set of errors.

I wonder what happens if `model_filename` doesn't exist, or isn't a zip, or cannot be read due to permission errors, or is a directory...all kinds of things could go wrong here. `get_model_metadata` seems to check for such problems by catching Exception and returning None in that case, but there is currently no unit test that verifies it (which also Codecov complains about).